src/main/java/frc/robot/subsystems/ShooterHoodSubsystem.java

```diff
@@ -0,0 +1,107 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANPIDController;
+import com.revrobotics.CANDigitalInput;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import frc.robot.Constants;
+
+public class ShooterHoodSubsystem extends SubsystemBase {
+  private CANSparkMax hoodNEO = new CANSparkMax(Constants.SPARKMAX_SHOOTER_HOOD_CAN_ADDR,
+      CANSparkMaxLowLevel.MotorType.kBrushless);
+  private CANDigitalInput hoodLimitSwitch = hoodNEO
+      .getReverseLimitSwitch(CANDigitalInput.LimitSwitchPolarity.kNormallyOpen);
+  private CANEncoder hoodEncoder = hoodNEO.getEncoder();
+  private CANPIDController hoodPID;
+
+  private final double kP = 5e-5;
+  private final double kI = 1e-6;
+  private final double kD = 0;
+  private final double kIz = 0;
+  private final double kFF = 0.000156;
+  private final double kMaxOutput = 0.3;
+  private final double kMinOutput = -0.3;
+  private final double allowedErr = 0;
```

Comment:
These are the constants for the full NEO.  I've not seen any published constants for the NEO550.  So we can start with these, but we'll likely have to tune them.